File: bots/rs/v55/src/builder/tasks/offense/scout_toward_enemy.rs
//! Translation of `bots/intgrah/v54.7.9/builder/tasks/offense/scout_toward_enemy.py`.
//!
//! Terminal OFFENSE fallback. Wraps `offense::helpers::scout_toward_enemy`
//! as a Task. Never rejects — guarantees the OFFENSE policy always produces
//! some movement, even when no harvester / cached target / conveyor target
//! is available.

use cambc::Controller;

use crate::builder::Builder;
use crate::builder::tasks::offense::helpers::scout_toward_enemy as scout;
use crate::builder::tasks::rejected::TaskResult;

pub fn scout_toward_enemy(self_: &mut Builder, ct: &mut Controller<'_>) -> TaskResult {
    scout(self_, ct);
    Ok(())
}

File: bots/beacon/bestrn/builder/tasks/offense/scout_toward_enemy.py
"""
Translation of `bots/intgrah/v54.7.9/builder/tasks/offense/scout_toward_enemy.py`.

Terminal OFFENSE fallback. Wraps `offense::helpers::scout_toward_enemy`
as a Task. Never rejects — guarantees the OFFENSE policy always produces
some movement, even when no harvester / cached target / conveyor target
is available.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from cambc import Controller
if TYPE_CHECKING:
    from builder import Builder
from builder.tasks.offense.helpers import scout_toward_enemy as scout

if TYPE_CHECKING:
    from builder.tasks.rejected import TaskResult


def scout_toward_enemy(self_, ct):
    scout(self_, ct)
    return None
